### PR TITLE
[Coinmate] add support for orderById

### PR DESCRIPTION
**Pull request by stachon**
Add support for getting order by id on Coinmate.io exchange

### CODE REVIEW

File: xchange-coinmate/src/main/java/org/knowm/xchange/coinmate/CoinmateAdapters.java [hunk 1]
```diff
@@ -338,4 +336,50 @@ public static String adaptSortOrder(TradeHistoryParamsSorted.Order order) {
         throw new IllegalArgumentException();
     }
   }
+
+  public static List<Order> apdaptOrders(CoinmateOrders coinmateOrders) {
```
Comment (earce) [inline xchange-coinmate/src/main/java/org/knowm/xchange/coinmate/CoinmateAdapters.java:340]: small typo in `adaptOrders`

Comment (stachon) [reply xchange-coinmate/src/main/java/org/knowm/xchange/coinmate/CoinmateAdapters.java]: Thanks, fixed

File: xchange-coinmate/src/main/java/org/knowm/xchange/coinmate/dto/trade/CoinmateOrder.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  private final String type;
+  private final BigDecimal price;
+  private final BigDecimal remainingAmount;
+  private final BigDecimal originalAmount;
+  private final BigDecimal stopPrice;
+  private final String status;
+  private final String orderTradeType;
+  private final BigDecimal avgPrice;
+
+  public CoinmateOrder(
```
Comment (earce) [inline xchange-coinmate/src/main/java/org/knowm/xchange/coinmate/dto/trade/CoinmateOrder.java:20]: should currency pair not be added into this as well?

File: xchange-coinmate/src/main/java/org/knowm/xchange/coinmate/CoinmateAdapters.java [hunk 3]
```diff
@@ -338,4 +336,50 @@ public static String adaptSortOrder(TradeHistoryParamsSorted.Order order) {
...
+            entry.getAvgPrice(),
+            entry.getOriginalAmount().subtract(entry.getRemainingAmount()),
+            null,
+            orderStatus,
+            null);
+
+
+    ordersList.add(order);
+
+    return ordersList;
```
Comment (earce) [inline xchange-coinmate/src/main/java/org/knowm/xchange/coinmate/CoinmateAdapters.java:383]: Not necessary but a bit more elegant you can do something like 

`return Collections.singletonList(new MarketOrder(orderType,
            entry.getOriginalAmount(),
            null,
            Long.toString(entry.getId()),
            new Date(entry.getTimestamp()),
            entry.getAvgPrice(),
            entry.getOriginalAmount().subtract(entry.getRemainingAmount()),
            null,
            orderStatus,
            null));`

File: xchange-coinmate/src/main/java/org/knowm/xchange/coinmate/CoinmateAdapters.java [hunk 4]
```diff
@@ -338,4 +336,50 @@ public static String adaptSortOrder(TradeHistoryParamsSorted.Order order) {
...
+      orderStatus = Order.OrderStatus.NEW;
+    } else {
+      orderStatus = Order.OrderStatus.UNKNOWN;
+    }
+
+    // TODO: we can probably use `orderTradeType` to distinguish between Market and Limit order
+    Order order =
+        new MarketOrder(orderType,
+            entry.getOriginalAmount(),
+            null,
```
[the remark this answers is not in the document]
Comment (stachon) [reply xchange-coinmate/src/main/java/org/knowm/xchange/coinmate/CoinmateAdapters.java:371]: Unfortunately the API does not provide currency pair here.

File: xchange-coinmate/src/main/java/org/knowm/xchange/coinmate/dto/trade/CoinmateOrder.java (hunk 2 above)
Comment (stachon) [reply xchange-coinmate/src/main/java/org/knowm/xchange/coinmate/dto/trade/CoinmateOrder.java:20]: Unfortunately the API does not provide currency pair here.

File: xchange-coinmate/src/main/java/org/knowm/xchange/coinmate/CoinmateAdapters.java (hunk 3 above)
Comment (stachon) [reply xchange-coinmate/src/main/java/org/knowm/xchange/coinmate/CoinmateAdapters.java:383]: Thanks

**Pull request merged**